packages/gitbook/src/app/(site)/(content)/[[...pathname]]/page.tsx

```diff
@@ -128,26 +144,28 @@ export async function generateViewport({ params }: { params: PagePathParams }):
     };
 }
 
-export async function generateMetadata({
-    params,
-    searchParams,
-}: {
-    params: PagePathParams;
-    searchParams: { fallback?: string };
-}): Promise<Metadata> {
-    const { space, pages, page, customization, site, ancestors } = await getPageDataWithFallback({
-        pagePathParams: params,
-        searchParams,
-    });
+function getTitle(input: Awaited<ReturnType<typeof getPageDataWithFallback>>) {
+    const { page, space, customization, site } = input;
+    return [page.title, getContentTitle(space, customization, site ?? null)]
+        .filter(Boolean)
+        .join(' | ');
+}
 
-    if (!page) {
-        notFound();
+export async function generateMetadata(props: PageProps): Promise<Metadata> {
+    const headerList = await headers();
+    const fetchMode = headerList.get('sec-fetch-mode');
+
+    // We only generate metadata in navigation mode. Else we let the browser handle it.
+    if (fetchMode !== 'navigate') {
```

Comment:
Are we sure bots for SEO will pass this header with this value?

Otherwise we'll lose all metadata

Reply:
Can you reverse the condition to be only equal to the case where we don't want it?